README.md

```diff
@@ -3,18 +3,281 @@
+	return "Hello " + req, nil
+}
+
+func (s *GreetServiceImpl) GreetWithStruct(ctx context.Context, req *hello.GreetRequest) (resp *hello.GreetResponse, err error) {
+	return &hello.GreetResponse{Resp: "Hello " + req.Req}, nil
+}
+```
+
+Implement interface in **handler.go**.
+
+#### initializing client
+
+```go
+import (
+	"context"
+	"github.com/cloudwego/kitex/client"
+	"github.com/cloudwego/kitex/pkg/klog"
+	dubbo "github.com/kitex-contrib/codec-dubbo/pkg"  
+	"github.com/kitex-contrib/codec-dubbo/samples/helloworld/kitex/kitex_gen/hello"
+	"github.com/kitex-contrib/codec-dubbo/samples/helloworld/kitex/kitex_gen/hello/greetservice"
+)
+
+func main() {
+	cli, err := greetservice.NewClient("helloworld",
+		// specify address of target server
+		client.WithHostPorts("127.0.0.1:21001"),
+		// configure dubbo codec
+		client.WithCodec(
+			dubbo.NewDubboCodec(
+				// target dubbo Interface Name
+				dubbo.WithJavaClassName("org.cloudwego.kitex.samples.api.GreetProvider"),
+			),
+		),
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	resp, err := cli.Greet(context.Background(), "world")
+	if err != nil {
+		klog.Error(err)
+		return
+	}
+	klog.Infof("resp: %s", resp)
+	
+	respWithStruct, err := cli.GreetWithStruct(context.Background(), &hello.GreetRequest{Req: "world"})
+	if err != nil {
+		klog.Error(err)
+		return
+	}
+	klog.Infof("respWithStruct: %s", respWithStruct.Resp)
+}
+```
+
+Important notes:
+1. Each dubbo Interface corresponds to a DubboCodec. Please do not configure multiple clients sharing a single DubboCodec.
+
+#### initializing server
 
-- [hessian serialization](http://hessian.caucho.com/doc/hessian-serialization.html)
+```go
+import (
+	"github.com/cloudwego/kitex/server"
+	dubbo "github.com/kitex-contrib/codec-dubbo/pkg"
+	hello "github.com/kitex-contrib/codec-dubbo/samples/helloworld/kitex/kitex_gen/hello/greetservice"
+	"log"
+	"net"
+)
+
+func main() {
+	// server address to listen on
+	addr, _ := net.ResolveTCPAddr("tcp", ":21000")
+	svr := hello.NewServer(new(GreetServiceImpl),
+		server.WithServiceAddr(addr),
+		// configure DubboCodec
+		server.WithCodec(dubbo.NewDubboCodec(
+			// Interface Name of kitex service. Other dubbo clients and kitex clients can refer to this name for invocation.
+			dubbo.WithJavaClassName("org.cloudwego.kitex.samples.api.GreetProvider"),
+		)),
+	)
+
+	err := svr.Run()
+
+	if err != nil {
+		log.Println(err.Error())
+	}
+}
+```
+
+Important notes:
+1. Each Interface Name corresponds to a DubboCodec. Please do not configure multiple servers sharing a single DubboCodec.
+
+## Benchmark
+
+### Benchmark Environment 
+CPU: **Intel(R) Xeon(R) Gold 5118 CPU @ 2.30GHz**  
+Memory: **192GB**
+
+### Benchmark Code
+
+Referring to [dubbo-go-benchmark](https://github.com/dubbogo/dubbo-go-benchmark). Converting dubbo client and dubbo server
+to kitex client and kitex server. Please see [this](https://github.com/kitex-contrib/codec-dubbo/tree/main/tests/benchmark).
+
+### Benchmark Result
+
+#### dubbo -> dubbo
```

Comment:
Better we don't mention dubbo-go here to avoid unnecessary arguments between two communities :)